src/main/java/com/opinta/service/DiscountPerCounterpartyServiceImpl.java

```diff
@@ -0,0 +1,177 @@
+        DiscountPerCounterparty discountPerCounterparty = discountPerCounterpartyMapper
+                .toEntity(discountPerCounterpartyDto);
+        discountPerCounterparty.setDiscount(discountService.getEntityByUuid(discountPerCounterpartyDto.getDiscountUuid()));
+        validateDiscountTime(discountPerCounterparty);
+        return discountPerCounterpartyMapper.toDto(saveEntity(discountPerCounterparty, user));
+    }
+    
+    private void validateDiscountTime(DiscountPerCounterparty discountPerCounterparty)
+            throws PerformProcessFailedException {
+        if (!discountPerCounterparty.isDiscountValidNow()) {
+            throw new PerformProcessFailedException(format(
+                    "DiscountPerCounterparty time from: %s to: %s is not valid!",
+                    dateFormat.format(discountPerCounterparty.getFromDate()),
+                    dateFormat.format(discountPerCounterparty.getToDate())));
+        }
+        if (!discountPerCounterparty.isDiscountPerCounterpartyValidForAssignedDiscount()) {
+            throw new PerformProcessFailedException(format(
+                    "DiscountPerCounterparty time from %s to: %s is not valid for assigned " +
+                            "Discount %s time from %s to: %s ",
+                    dateFormat.format(discountPerCounterparty.getFromDate()),
+                    dateFormat.format(discountPerCounterparty.getToDate()),
+                    discountPerCounterparty.getDiscount().getUuid().toString(),
+                    dateFormat.format(discountPerCounterparty.getDiscount().getFromDate()),
+                    dateFormat.format(discountPerCounterparty.getDiscount().getToDate())));
+        }
+    }
+
+    @Override
+    @Transactional
+    public DiscountPerCounterpartyDto update(UUID uuid, DiscountPerCounterpartyDto discountPerCounterpartyDto,
+                                             User user) throws IncorrectInputDataException, AuthException,
+            PerformProcessFailedException {
+        DiscountPerCounterparty source = discountPerCounterpartyMapper.toEntity(discountPerCounterpartyDto);
+        source.setDiscount(discountService.getEntityByUuid(source.getDiscount().getUuid()));
+        DiscountPerCounterparty target = getEntityByUuid(uuid, user);
+        validateDiscountTime(source);
+        source.setCounterparty(target.getCounterparty());
+
+        try {
+            copyNotNullProperties(target, source);
+        } catch (Exception e) {
+            log.error(copyPropertiesOnErrorLogEndpoint(DiscountPerCounterparty.class, source, target, e));
+            throw new PerformProcessFailedException(copyPropertiesOnErrorLogEndpoint(
+                    DiscountPerCounterparty.class, source, target, e));
+        }
+        target.setUuid(uuid);
+        log.info(updateLogEndpoint(DiscountPerCounterparty.class, target));
+        discountPerCounterpartyDao.update(target);
```

Comment:
Plz, use updateEntity() instead
No need of all this code, move it to updateEntity, like in DiscountServiceImpl.update() done